.envrc

```diff
@@ -1,26 +1,37 @@
-# Build a folder ./.env that contains all the tools
+# Build a folder ./.env that contains the dev-shell which load all the right env
+# vars.
+#
 # To speed up the nix evaluation, we only rebuild our environment when `./nix`
-# changes. We do this by adding its contents to the nix store and using its
-# store path as a cache key. We only use this to build the dev environment. This
-# will fail building in case anything inside the devEnv is reaching outside
-# the nix/ subfolder
-store_path=$(nix-store --add ./nix/)
+# or any of the `default.nix` files change. We do this by adding all these files
+# to the nix store and using the store paths as a cache key.
+
+store_paths=$(find . -name default.nix  | grep -v '^./nix' | grep -v '^./dist-newstyle' | xargs nix-store --add ./nix)
 layout_dir=$(direnv_layout_dir)
+env_dir=./.env
+
+# The env_dir used to be a symlink to env in nix store, if it exists, delete it.
+[[ ! -L "$env_dir" ]] || rm -f "$env_dir"
+
 [[ -d "$layout_dir" ]] || mkdir -p "$layout_dir"
-if [[ ! -d ./.env || ! -f "$layout_dir/nix-rebuild" || "$store_path" != $(< "$layout_dir/nix-rebuild" ) ]]; then
 
-  # this builds "devEnv" defined in ./nix/default.nix and outputs to ./.env
-  nix-build "$store_path" -A devEnv --out-link ./.env
+if [[ ! -d "$env_dir" || ! -f "$layout_dir/nix-rebuild" || "$store_paths" != $(< "$layout_dir/nix-rebuild" ) ]]; then
+
+  [[ -d "$env_dir" ]] || mkdir -p "$env_dir"
+
+  # this builds "wireServer.devShell" defined in ./nix/default.nix and outputs to ./.env
+  nix print-dev-env --impure --expr "(import ./nix).wireServer.devShell" > "${env_dir}/dev-shell.sh"
```

Comment:
I'm using nix 2.3 (from Debian Stable's nix package), which doesn't have a `print-dev-env`. Are you able to implement this bit of the script without using this nix feature, or do we need to set a minimum bound on the version of nix which we require in `wire-server`?

Reply:
Hmm, I copied this out of direnv's `use_nix` function. Let me see if they have some old ways of doing the same thing.